sdk/python/lib/pulumi/resource.py

```diff
@@ -865,9 +875,11 @@ def export(name: str, value: Any):
     :param str name: The name to assign to this output.
     :param Any value: The value of this output.
     """
-    stack = get_root_resource()
-    if stack is not None:
-        stack.output(name, value)
+    res = cast('Stack', get_root_resource())
```

Comment:
interesting.  waht is get_root_resource typed to return if not Stack?

Reply:
sounds lke this cast should technically move below the `is_stack` line.

Reply:
`Resource` - though we could/should fix that there in follow-up to reduce use of `cast`.

Reply:
`Optional[Resource]` https://github.com/pulumi/pulumi/blob/c49a5c1e73a0cd8d3f81236ebf639dccb635e64b/sdk/python/lib/pulumi/runtime/settings.py#L176